src/Factory/FixtureInstantiator/FixtureInstantiatorForParametrizedConstructors.php

```diff
@@ -25,27 +29,29 @@ public function supports(string $fixtureClass): bool
 
     public function instantiate(string $fixtureClass): Fixture
     {
-        $reflectionClass = new \ReflectionClass($fixtureClass);
-        $constructor = $reflectionClass->getConstructor();
+        $constructor = (new \ReflectionClass($fixtureClass))->getConstructor();
+        \assert($constructor instanceof \ReflectionMethod);
 
         $constructorServices = [];
         foreach ($constructor->getParameters() as $parameter) {
             $reflectionType = $parameter->getType();
 
-            if (!$reflectionType) {
+            if (!$reflectionType instanceof \ReflectionNamedType) {
                 $this->throwLogicException(
                     'Parameter "$%s" of %s::%s() has no type, while it should be an implementation of "%s".',
                     $parameter,
                 );
             }
 
-            $reflectionTypeName = $reflectionType->getName();
-            if (ContainerInterface::class === $reflectionTypeName) {
+            $type = $reflectionType->getName();
+
+            if (ContainerInterface::class === $type) {
                 $constructorServices[] = $this->container;
                 continue;
             }
-            if ($this->container->has($reflectionTypeName)) {
-                $constructorServices[] = $this->container->get($reflectionTypeName);
+
+            if ($this->container->has($type)) {
```

Comment:
Just to mention: This will not work with future Symfony and PHPUnit version. The Container is a special TestContainer and does not have the ContainerInterface as service. But it's possible to pass the TestContainer itself as argument.

Reply:
Right, that's why we handle it explicitly above.